datasets/noaa/_images/run_csv_transform_kub_gsod_stations/csv_transform.py

```diff
@@ -0,0 +1,304 @@
+        df["name"][:].replace("^\\s{1,}([a-zA-Z]\\D+)", "$1", regex=True, inplace=True)
+        df["name"][:].replace("^(\\D+[a-zA-Z])\\s{1,}$", "$1", regex=True, inplace=True)
+        df["name"][:].replace("^(\\s+)$", "", regex=True, inplace=True)
+
+        curr_dtm = str(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
+        logging.info(
+            f"           Executing Reg Ex.. (call) {source_file} at {curr_dtm}"
+        )
+        df["call"][:].replace("^(\\s+)$", "", regex=True, inplace=True)
+        df["call"][:].replace("^([a-zA-Z]+)\\s+", "$1", regex=True, inplace=True)
+
+        curr_dtm = str(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
+        logging.info(
+            f"           Executing Reg Ex.. (elev) {source_file} at {curr_dtm}"
+        )
+        df["elev"][:].replace("^(\\s+)$", "", regex=True, inplace=True)
+
+        curr_dtm = str(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
+        logging.info(
+            f"           Executing Reg Ex.. (state) {source_file} at {curr_dtm}"
+        )
+        df["state"][:].replace("^(\\s+)$", "", regex=True, inplace=True)
+
+        curr_dtm = str(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
+        logging.info(
+            f"           Executing Reg Ex.. (country) {source_file} at {curr_dtm}"
+        )
+        df["country"][:].replace("^(\\s+)$", "", regex=True, inplace=True)
+
+        curr_dtm = str(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
+        logging.info(f"Transform: Saving to output file.. {target_file} at {curr_dtm}")
+        df.to_csv(target_file, index=False)
+
+        curr_dtm = str(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
+        logging.info(f"completed processing {source_url} at {curr_dtm}")
+
+        curr_dtm = str(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
+        logging.info(
+            f"Uploading output file to.. gs://{target_gcs_bucket}/{target_gcs_path} at {curr_dtm}"
+        )
+        upload_file_to_gcs(target_file, target_gcs_bucket, target_gcs_path)
+
+        curr_dtm = str(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
+        logging.info(f"NOAA GSOD Stations process completed at {curr_dtm}")
+
+    else:
+
+        # we were unable to reach the url
+        curr_dtm = str(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
+        logging.info(f"Error: Unable to reach url: {source_url} at {curr_dtm}")
+        logging.info("Process failed!")
+
+
+def replace_value(val: str) -> str:
+    if val is None or len(val) == 0:
+        return val
+    else:
+        if val.find("\n") > 0:
+            # return val.replace("\n", "")
+            return re.sub(r"(^\d):(\d{2}:\d{2})", "0$1:$2", val)
+        else:
+            return val
+
+
+def replace_values_regex(df: pd.DataFrame) -> None:
+    header_names = {"checkout_time"}
+
+    for dt_col in header_names:
+        if df[dt_col] is not None:
+            if df[dt_col].str.len() > 0:
```

Comment:
You can combine into`df[dt_col] is not None && df[dt_col].str:`

Reply:
Done